### DISCUSSION TITLE
Connection-specific VPN settings

### DISCUSSION THREAD

**Discussion by wisp3rwind** [mullvad/mullvadvpn-app#9103] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

I'd appreciate if Mullvad would offer **connection-specific overrides for all/most VPN settings**. Specifically, I would have had a use for this with

- local network sharing (should only be enabled in trusted networks where I have a need for it, e.g. at home or maybe at a friend's place)
- UDP-over-TCP (only required for networks which block UDP, e.g. _some_ public Wifis, but not my home network)

but I think it makes sense for several other settings as well.

Currently, anytime the necessity to change these settings (away from a secure and performant default) arises, I'll usually proceed via:

- some debugging of the present networking issues until realizing it's the VPN,
- a bunch of clicks to change settings,
- use the network,
- frequently forget to change the settings back (which might compromise security or performance when connecting to another network).

By making such setting adaptations per-network, these issues would be remedied to a large extent, for a better UX and effectively improved security.

In #3218, some concerns regarding security have been voiced. However, at least in my case, the real concern is insecure public Wifis rather than someone spoofing a BSSID I trust.
Anyone who actually fears a targeted spoofing attack can simply not use connection-specific settings, and appropriate warnings in the UI could create the awareness for this.

Some further ideas to in this regard:
- there should be an interface to review connection-specific settings at a glance, making it easy to revoke them when deemed outdated
- one could consider expiry timers (i.e. disable a connection-specific setting when it has not been used in a certain amount of time)

Related feature requests are:
- https://github.com/mullvad/mullvadvpn-app/issues/7133 "Allow to turn off features from the main view" (This would improve the UX; but still require manual action.)
- https://github.com/mullvad/mullvadvpn-app/issues/7134 "Redirect to corresponding view when clicking feature labels on the main view" (similar to the previous issue)
- https://github.com/mullvad/mullvadvpn-app/issues/3218 "Add the ability to trust a network for local network sharing" (this is pretty close the current feature request, but I'd argue that this should cover VPN settings in general, not just local sharing)

### Alternative solutions

I don't think there's a real alternative: This really needs to be as automated as possible, otherwise, I'll inevitably forget to change settings back. Usually, I realize this only a long time afterwards, for example when seeing unexpected network devices discovered by the file manager (i.e. too late).

### Type of feature

- [X] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [X] Easier to use
- [ ] Other

### Operating System

- [X] Android
- [X] iOS
- [X] Windows
- [X] macOS
- [X] Linux

**Comment by MarkusPettersson98** [comment]
Hello! Thanks for your detailed feature request! :raised_hands: 

We'll keep an eye on the interest from the community on this. We know that similar feature requests have been made before, so it's not an esoteric request.

**Comment by musabagriyanik** [comment]
I'm also looking for this, and there is a real use case for that. WireGuard is blocked in my cellular network so I use Shadowsocks, but my other networks that doesn't block WireGuard also has to use Shadowsocks unless I disable it, which is not something you'd want to do every time you're going to a place and leaving. There is a performance loss happening due to Shadowsocks, it's small but it's there. So we should be able to customize connection details for specific conditions.

**Comment by Aeldit** [comment]
Hi, I've been looking for this feature too: in my home, I have some git repositories hosted locally only and some servers I can connect to using ssh. Having a per-network setting would be really helpful for these cases!

**Comment by Nrwh1** [comment]
Definitely interested in this. This just makes sense to have and would love to see it implemented.

**Comment by thomaswitt** [comment]
This is a real blocker for me as well.

I am not asking for silent auto-trust of networks. I am asking for explicit, opt-in, connection-specific overrides with strong warnings and a persistent “less private settings active” indicator.

My concrete use case:
- home Ethernet / home SSID: allow local network sharing, possibly different transport settings
- all other networks: strict secure defaults

For me, the current manual toggle is actually riskier, because I can forget to revert it after using it at home. I understand the spoofing caveat. I would still prefer this as an advanced opt-in feature with clear warnings over the current all-manual workflow.
